src/app/rooms/page.tsx

```diff
@@ -1,162 +1,194 @@
+      const updatedGames = [...prevGames];
+      if (index !== -1) {
+        updatedGames.splice(index, 1, game);
+      } else {
+        updatedGames.push(game);
+      }
+      return updatedGames;
     });
+  };
 
-    socket.on('onParticipateGame', (game: any) => {
-      setGames(prevGames => {
-        const indexOfGame = prevGames.findIndex(
-          curGame => curGame.id === game.id,
-        );
-        const copyOfGames = [...prevGames];
-        copyOfGames.splice(indexOfGame, 1, game);
-        return copyOfGames;
-      });
+  const fetchChatData = async () => {
+    const chatData = await socket.emitWithAck('chatData', {
+      chatId: MUTUAL_CHAT_ID,
     });
+    setMessages(chatData.messages);
+  };
 
-    async function getChatData() {
-      const chatData = await socket.emitWithAck('chatData', {
-        chatId: MUTUAL_CHAT_ID,
-      });
-      setMessages(chatData.messages);
-    }
-    getChatData();
+  const fetchGames = async () => {
+    const games = await socket.emitWithAck('getVisibleGames');
+    setGames(games);
+  };
 
+  useEffect(() => {
+    // Register socket event listeners
+    socket.on('onMessage', handleOnMessage);
+    socket.on('clearStartedGame', handleClearStartedGame);
+    socket.on('startGame', handleStartGame);
+    socket.on('onParticipateGame', handleOnParticipateGame);
+    socket.on('error', console.error);
+
+    // Fetch initial chat data
+    fetchChatData();
+    fetchGames();
+
+    // Cleanup listeners on component unmount
     return () => {
-      socket.off('onMessage');
+      socket.off('onMessage', handleOnMessage);
+      socket.off('clearStartedGame', handleClearStartedGame);
+      socket.off('startGame', handleStartGame);
+      socket.off('onParticipateGame', handleOnParticipateGame);
+      socket.off('connect');
+      socket.off('disconnect');
       socket.off('error');
     };
-  }, []);
+  }, [router]);
 
   const sendMessage = () => {
-    if (socket) {
-      socket.emit('newMessage', {
-        text: message.text,
-        chatId: MUTUAL_CHAT_ID,
-      });
-      setMessage({ text: '', chatId: MUTUAL_CHAT_ID });
+    socket.emit('newMessage', { text: message.text, chatId: MUTUAL_CHAT_ID });
+    setMessage(prevMessage => ({ ...prevMessage, text: '' }));
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      sendMessage();
     }
   };
 
   const handleJoinGame = (id: string) => {
-    socket?.emit('joinGame', { id });
+    socket.emit('joinGame', { id });
+  };
+
+  const handleLeaveGame = (id: string) => {
+    socket.emit('leaveGame', { id });
   };
 
-  const hadleLeaveGame = (id: string) => {
-    socket?.emit('leaveGame', { id });
+  const messagesEndRef = useRef<HTMLDivElement | null>(null);
+
+  const scrollToBottom = () => {
+    if (messagesEndRef.current) {
+      messagesEndRef.current.scrollIntoView({ behavior: 'instant' });
+    }
   };
+
+  useEffect(() => {
```

Comment:
if user scrolled to the top and a new message has been sent, wouldn`t it be weird, to scroll against user to the bottom?